modules/Xsens_module/src/XsensRetargeting.cpp

```diff
@@ -0,0 +1,287 @@
+    {
+        yError() << "[XsensRetargeting::configure] Unable to find human_joint_list_stream into "
+                    "config file.";
+        return false;
+    }
+
+    if (!YarpHelper::yarpListToStringVector(humanAxesListYarp, m_humanJointsListName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to convert yarp list into a "
+                    "vector of strings.";
+        return false;
+    }
+
+    yInfo() << "Human joints name list: [human joints list] [robot joints list]"
+            << m_humanJointsListName.size() << " , " << m_robotJointsListNames.size();
+
+    for (size_t i = 0; i < m_humanJointsListName.size(); i++)
+    {
+        if (i < m_robotJointsListNames.size())
+            yInfo() << "(" << i << "): " << m_humanJointsListName[i] << " , "
+                    << m_robotJointsListNames[i];
+        else
+        {
+            yInfo() << "(" << i << "): " << m_humanJointsListName[i] << " , --";
+        }
+    }
+
+    std::string portName;
+    if (!YarpHelper::getStringFromSearchable(rf, "wholeBodyJointsPort", portName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to get a string from a searchable";
+        return false;
+    }
+    if (!m_wholeBodyHumanJointsPort.open("/" + getName() + portName))
+    {
+        yError() << "[XsensRetargeting::configure] " << portName << " port already open.";
+        return false;
+    }
+
+    if (!YarpHelper::getStringFromSearchable(rf, "controllerPort", portName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to get a string from a searchable";
+        return false;
+    }
+    if (!m_wholeBodyHumanSmoothedJointsPort.open("/" + getName() + portName))
+    {
+        yError() << "[XsensRetargeting::configure] Unable to open the port " << portName;
+        return false;
+    }
+
+    // I should do a maping between two vectors here.
+
+    bool foundMatch = false;
+    for (unsigned i = 0; i < m_robotJointsListNames.size(); i++)
+    {
+        for (unsigned j = 0; j < m_humanJointsListName.size(); j++)
+        {
+
+            if (m_robotJointsListNames[i] == m_humanJointsListName[j])
+            {
+                foundMatch = true;
+                m_humanToRobotMap.push_back(j);
+                break;
+            }
+        }
+        if (!foundMatch)
+        {
+            yError() << "not found match for: " << m_robotJointsListNames[i] << " , " << i;
+            ;
+            return false;
+        }
+        foundMatch = false;
+    }
+
+    yInfo() << "*** mapped joint names: ****";
+    for (size_t i = 0; i < m_robotJointsListNames.size(); i++)
+    {
+        yInfo() << "(" << i << ", " << m_humanToRobotMap[i] << "): " << m_robotJointsListNames[i]
+                << " , " << m_humanJointsListName[(m_humanToRobotMap[i])];
+    }
+
+    //     m_tick =std::chrono::duration_cast< std::chrono::milliseconds
+    //     >(std::chrono::system_clock::now().time_since_epoch()); m_tock
```

Comment:
If you want to use a time profiler you can use [the one in `walking-controllers`](https://github.com/robotology/walking-controllers/blob/master/modules/Walking_module/src/TimeProfiler.cpp). Actually there are several utilities in walking-controllers that can be used in different scenarios (e.g.  [`Time-profiler`](https://github.com/robotology/walking-controllers/blob/master/modules/Walking_module/src/TimeProfiler.cpp), [Utilities library ](https://github.com/robotology/walking-controllers/tree/master/modules/Utilities)). Probably we should find a way (e.g. move them in an other repo) to expose these libraries. @traversaro

Reply:
We can either move them in a different repo, or start installing some of the code contained in walking-controllers as libraries (that may be convenenient if we eventually implement https://github.com/robotology/walking-controllers/issues/30 ).